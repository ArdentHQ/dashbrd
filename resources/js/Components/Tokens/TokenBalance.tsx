import { useRef } from "react";
import { useTranslation } from "react-i18next";
import { TokenActions } from "@/Components/Tokens/TokenActions";
import { Tooltip } from "@/Components/Tooltip";
import { useIsTruncated } from "@/Hooks/useIsTruncated";
import { formatCrypto, formatFiat } from "@/Utils/Currency";
import { FormatPercentage } from "@/Utils/Percentage";

interface Properties {
    asset: App.Data.TokenListItemData;
    currency: string;
    onSend?: () => void;
    onReceive?: () => void;
}

export const TokenBalance = ({ asset, currency, onSend, onReceive }: Properties): JSX.Element => {
    const { t } = useTranslation();

    const token: Pick<App.Data.Token.TokenData, "symbol" | "name" | "decimals"> = {
        symbol: asset.symbol,
        name: asset.name,
        decimals: asset.decimals,
    };
    const formattedBalance = formatCrypto({
        value: asset.balance.toString(),
        token,
        t,
    });

    const fiatBalance = formatFiat({
        value: asset.fiat_balance ?? "0",
        currency,
        t,
    });

    const cryptoBalanceReference = useRef<HTMLSpanElement>(null);
    const isCryptoBalanceTruncated = useIsTruncated({ reference: cryptoBalanceReference });

    const fiatBalanceReference = useRef<HTMLParagraphElement>(null);
    const isFiatBalanceTruncated = useIsTruncated({ reference: fiatBalanceReference });

    return (
        <div className="flex flex-col space-y-4 px-6 py-6 sm:px-8">
            <div className="flex flex-col items-center space-y-4 sm:flex-row sm:items-end sm:space-y-0">
                <div className="w-full sm:w-2/3">
                    <div className="space-y-0.5 text-center font-medium sm:text-left">
                        <p className="text-sm leading-5.5 text-theme-secondary-500 dark:text-theme-dark-400">
                            {t("common.my_balance")}
                        </p>
                        <Tooltip
                            content={fiatBalance}
                            disabled={!isFiatBalanceTruncated}
                        >
                            <p
                                className="truncate text-2xl text-theme-secondary-900 dark:text-theme-dark-50"
                                ref={fiatBalanceReference}
                            >
                                {fiatBalance}
                            </p>
                        </Tooltip>
                    </div>
                </div>

                <div className="sm:w-1/3">
                    <TokenActions
                        balance={asset.balance}
                        onSend={onSend}
                        onReceive={onReceive}
                    />
                </div>
            </div>

            <div className="flex flex-col space-y-3">
                <div className="flex justify-between text-sm font-medium">
                    <Tooltip
                        content={`${formattedBalance} ${asset.symbol}`}
                        disabled={!isCryptoBalanceTruncated}
                    >
                        <div className="flex min-w-0 pr-2 text-theme-secondary-900 dark:text-theme-dark-50">
                            <span
                                className="mr-1 truncate"
                                ref={cryptoBalanceReference}
                            >
                                {formattedBalance}
                            </span>
                            <span className="dark:text-theme-dark-300">{asset.symbol}</span>
                        </div>
                    </Tooltip>

                    <Tooltip content={t("pages.token_panel.balance_tooltip")}>
                        <span className="text-theme-secondary-700 dark:text-theme-dark-200">
                            <FormatPercentage
                                value={Number(asset.percentage)}
                                decimals={2}
                            />
                        </span>
                    </Tooltip>
                </div>

                <div className="h-2 bg-theme-primary-200 dark:bg-theme-dark-600">
                    <div
                        className="h-2 bg-theme-primary-600"
                        style={{
                            width: `${Math.round((Number(asset.percentage) + Number.EPSILON) * 100)}%`,
                        }}
                    ></div>
                </div>
            </div>
        </div>
    );
};
