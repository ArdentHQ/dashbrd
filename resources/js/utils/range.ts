/**
 * Create a range of numbers between the two values.
 * If `end` is omitted, then the function will create a range of numbers from 0 to `start` (exluding the `start`).
 *
 * `range(5)` => [0, 1, 2, 3, 4]
 *
 * `range(4, 8)` => [4, 5, 6, 7]
 */
export const range = (start: number, end?: number): number[] => {
    if (end === undefined) {
        return Array.from({
            length: start,
        }).map((_, index) => index);
    }

    return Array.from({
        length: end - start,
    })
        .map((_, index) => index)
        .map((index) => index + start);
};
