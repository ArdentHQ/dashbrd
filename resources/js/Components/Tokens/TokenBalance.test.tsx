import React from "react";
import { TokenBalance } from "./TokenBalance";
import TokenListItemDataFactory from "@/Tests/Factories/Token/TokenListItemDataFactory";
import { render, screen } from "@/Tests/testing-library";
import { allBreakpoints } from "@/Tests/utils";

describe("TokenBalance", () => {
    it.each(allBreakpoints)("should render in %s screen", (breakpoint) => {
        const asset = new TokenListItemDataFactory().create({
            chain_id: 1,
            balance: (123 * 1e18).toString(),
            fiat_balance: "100",
            address: "0x123",
            symbol: "TEST",
            name: "Testing",
            decimals: 18,
            percentage: "0.5",
        });

        const { asFragment } = render(
            <TokenBalance
                asset={asset}
                currency="USD"
            />,
            { breakpoint },
        );

        expect(screen.getByText("$100.00")).toBeTruthy();
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render 0 if an undefined fiat balance is provided", () => {
        const asset = new TokenListItemDataFactory().create({
            chain_id: 1,
            balance: (123 * 1e18).toString(),
            address: "0x123",
            fiat_balance: undefined,
            symbol: "TEST",
            name: "Testing",
            decimals: 18,
            percentage: "0.5",
        });

        const { asFragment } = render(
            <TokenBalance
                asset={asset}
                currency="USD"
            />,
        );

        expect(screen.getByText("$0.00")).toBeTruthy();
        expect(asFragment()).toMatchSnapshot();
    });
});
