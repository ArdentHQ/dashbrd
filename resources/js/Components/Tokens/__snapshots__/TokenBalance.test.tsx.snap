// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TokenBalance > should render 0 if an undefined fiat balance is provided 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col space-y-4 px-6 py-6 sm:px-8"
  >
    <div
      class="flex flex-col items-center space-y-4 sm:flex-row sm:items-end sm:space-y-0"
    >
      <div
        class="w-full sm:w-2/3"
      >
        <div
          class="space-y-0.5 text-center font-medium sm:text-left"
        >
          <p
            class="text-sm leading-5.5 text-theme-secondary-500 dark:text-theme-dark-400"
          >
            My Balance
          </p>
          <p
            class="truncate text-2xl text-theme-secondary-900 dark:text-theme-dark-50"
          >
            $0.00
          </p>
        </div>
      </div>
      <div
        class="sm:w-1/3"
      >
        <div
          class="flex flex-shrink-0 items-center justify-end space-x-3"
        >
          <span
            data-testid="TokenActions__send"
            tabindex="-1"
          >
            <button
              aria-label="Send"
              class="group button-icon-primary"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowUp"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.9 19V9.9h3.8c.3 0 .4-.2.2-.5l-5.7-8.3c-.1-.1-.3-.1-.4 0L4.1 9.4c-.2.4-.1.6.2.6h3.8v3.1"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
          <span
            data-testid="TokenActions__receive"
            tabindex="-1"
          >
            <button
              aria-label="Receive"
              class="group button-icon"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowDown"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.104 1v9.1h-3.8c-.3 0-.4.2-.2.5l5.7 8.3c.1.1.3.1.4 0l5.7-8.3c.2-.4.1-.6-.2-.6h-3.8V6.9"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col space-y-3"
    >
      <div
        class="flex justify-between text-sm font-medium"
      >
        <div
          class="flex min-w-0 pr-2 text-theme-secondary-900 dark:text-theme-dark-50"
        >
          <span
            class="mr-1 truncate"
          >
            123
          </span>
          <span
            class="dark:text-theme-dark-300"
          >
            TEST
          </span>
        </div>
        <span
          class="text-theme-secondary-700 dark:text-theme-dark-200"
        >
          50%
        </span>
      </div>
      <div
        class="h-2 bg-theme-primary-200 dark:bg-theme-dark-600"
      >
        <div
          class="h-2 bg-theme-primary-600"
          style="width: 50%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TokenBalance > should render in lg screen 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col space-y-4 px-6 py-6 sm:px-8"
  >
    <div
      class="flex flex-col items-center space-y-4 sm:flex-row sm:items-end sm:space-y-0"
    >
      <div
        class="w-full sm:w-2/3"
      >
        <div
          class="space-y-0.5 text-center font-medium sm:text-left"
        >
          <p
            class="text-sm leading-5.5 text-theme-secondary-500 dark:text-theme-dark-400"
          >
            My Balance
          </p>
          <p
            class="truncate text-2xl text-theme-secondary-900 dark:text-theme-dark-50"
          >
            $100.00
          </p>
        </div>
      </div>
      <div
        class="sm:w-1/3"
      >
        <div
          class="flex flex-shrink-0 items-center justify-end space-x-3"
        >
          <span
            data-testid="TokenActions__send"
            tabindex="-1"
          >
            <button
              aria-label="Send"
              class="group button-icon-primary"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowUp"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.9 19V9.9h3.8c.3 0 .4-.2.2-.5l-5.7-8.3c-.1-.1-.3-.1-.4 0L4.1 9.4c-.2.4-.1.6.2.6h3.8v3.1"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
          <span
            data-testid="TokenActions__receive"
            tabindex="-1"
          >
            <button
              aria-label="Receive"
              class="group button-icon"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowDown"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.104 1v9.1h-3.8c-.3 0-.4.2-.2.5l5.7 8.3c.1.1.3.1.4 0l5.7-8.3c.2-.4.1-.6-.2-.6h-3.8V6.9"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col space-y-3"
    >
      <div
        class="flex justify-between text-sm font-medium"
      >
        <div
          class="flex min-w-0 pr-2 text-theme-secondary-900 dark:text-theme-dark-50"
        >
          <span
            class="mr-1 truncate"
          >
            123
          </span>
          <span
            class="dark:text-theme-dark-300"
          >
            TEST
          </span>
        </div>
        <span
          class="text-theme-secondary-700 dark:text-theme-dark-200"
        >
          50%
        </span>
      </div>
      <div
        class="h-2 bg-theme-primary-200 dark:bg-theme-dark-600"
      >
        <div
          class="h-2 bg-theme-primary-600"
          style="width: 50%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TokenBalance > should render in md screen 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col space-y-4 px-6 py-6 sm:px-8"
  >
    <div
      class="flex flex-col items-center space-y-4 sm:flex-row sm:items-end sm:space-y-0"
    >
      <div
        class="w-full sm:w-2/3"
      >
        <div
          class="space-y-0.5 text-center font-medium sm:text-left"
        >
          <p
            class="text-sm leading-5.5 text-theme-secondary-500 dark:text-theme-dark-400"
          >
            My Balance
          </p>
          <p
            class="truncate text-2xl text-theme-secondary-900 dark:text-theme-dark-50"
          >
            $100.00
          </p>
        </div>
      </div>
      <div
        class="sm:w-1/3"
      >
        <div
          class="flex flex-shrink-0 items-center justify-end space-x-3"
        >
          <span
            data-testid="TokenActions__send"
            tabindex="-1"
          >
            <button
              aria-label="Send"
              class="group button-icon-primary"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowUp"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.9 19V9.9h3.8c.3 0 .4-.2.2-.5l-5.7-8.3c-.1-.1-.3-.1-.4 0L4.1 9.4c-.2.4-.1.6.2.6h3.8v3.1"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
          <span
            data-testid="TokenActions__receive"
            tabindex="-1"
          >
            <button
              aria-label="Receive"
              class="group button-icon"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowDown"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.104 1v9.1h-3.8c-.3 0-.4.2-.2.5l5.7 8.3c.1.1.3.1.4 0l5.7-8.3c.2-.4.1-.6-.2-.6h-3.8V6.9"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col space-y-3"
    >
      <div
        class="flex justify-between text-sm font-medium"
      >
        <div
          class="flex min-w-0 pr-2 text-theme-secondary-900 dark:text-theme-dark-50"
        >
          <span
            class="mr-1 truncate"
          >
            123
          </span>
          <span
            class="dark:text-theme-dark-300"
          >
            TEST
          </span>
        </div>
        <span
          class="text-theme-secondary-700 dark:text-theme-dark-200"
        >
          50%
        </span>
      </div>
      <div
        class="h-2 bg-theme-primary-200 dark:bg-theme-dark-600"
      >
        <div
          class="h-2 bg-theme-primary-600"
          style="width: 50%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TokenBalance > should render in md-lg screen 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col space-y-4 px-6 py-6 sm:px-8"
  >
    <div
      class="flex flex-col items-center space-y-4 sm:flex-row sm:items-end sm:space-y-0"
    >
      <div
        class="w-full sm:w-2/3"
      >
        <div
          class="space-y-0.5 text-center font-medium sm:text-left"
        >
          <p
            class="text-sm leading-5.5 text-theme-secondary-500 dark:text-theme-dark-400"
          >
            My Balance
          </p>
          <p
            class="truncate text-2xl text-theme-secondary-900 dark:text-theme-dark-50"
          >
            $100.00
          </p>
        </div>
      </div>
      <div
        class="sm:w-1/3"
      >
        <div
          class="flex flex-shrink-0 items-center justify-end space-x-3"
        >
          <span
            data-testid="TokenActions__send"
            tabindex="-1"
          >
            <button
              aria-label="Send"
              class="group button-icon-primary"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowUp"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.9 19V9.9h3.8c.3 0 .4-.2.2-.5l-5.7-8.3c-.1-.1-.3-.1-.4 0L4.1 9.4c-.2.4-.1.6.2.6h3.8v3.1"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
          <span
            data-testid="TokenActions__receive"
            tabindex="-1"
          >
            <button
              aria-label="Receive"
              class="group button-icon"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowDown"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.104 1v9.1h-3.8c-.3 0-.4.2-.2.5l5.7 8.3c.1.1.3.1.4 0l5.7-8.3c.2-.4.1-.6-.2-.6h-3.8V6.9"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col space-y-3"
    >
      <div
        class="flex justify-between text-sm font-medium"
      >
        <div
          class="flex min-w-0 pr-2 text-theme-secondary-900 dark:text-theme-dark-50"
        >
          <span
            class="mr-1 truncate"
          >
            123
          </span>
          <span
            class="dark:text-theme-dark-300"
          >
            TEST
          </span>
        </div>
        <span
          class="text-theme-secondary-700 dark:text-theme-dark-200"
        >
          50%
        </span>
      </div>
      <div
        class="h-2 bg-theme-primary-200 dark:bg-theme-dark-600"
      >
        <div
          class="h-2 bg-theme-primary-600"
          style="width: 50%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TokenBalance > should render in sm screen 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col space-y-4 px-6 py-6 sm:px-8"
  >
    <div
      class="flex flex-col items-center space-y-4 sm:flex-row sm:items-end sm:space-y-0"
    >
      <div
        class="w-full sm:w-2/3"
      >
        <div
          class="space-y-0.5 text-center font-medium sm:text-left"
        >
          <p
            class="text-sm leading-5.5 text-theme-secondary-500 dark:text-theme-dark-400"
          >
            My Balance
          </p>
          <p
            class="truncate text-2xl text-theme-secondary-900 dark:text-theme-dark-50"
          >
            $100.00
          </p>
        </div>
      </div>
      <div
        class="sm:w-1/3"
      >
        <div
          class="flex flex-shrink-0 items-center justify-end space-x-3"
        >
          <span
            data-testid="TokenActions__send"
            tabindex="-1"
          >
            <button
              aria-label="Send"
              class="group button-icon-primary"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowUp"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.9 19V9.9h3.8c.3 0 .4-.2.2-.5l-5.7-8.3c-.1-.1-.3-.1-.4 0L4.1 9.4c-.2.4-.1.6.2.6h3.8v3.1"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
          <span
            data-testid="TokenActions__receive"
            tabindex="-1"
          >
            <button
              aria-label="Receive"
              class="group button-icon"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowDown"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.104 1v9.1h-3.8c-.3 0-.4.2-.2.5l5.7 8.3c.1.1.3.1.4 0l5.7-8.3c.2-.4.1-.6-.2-.6h-3.8V6.9"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col space-y-3"
    >
      <div
        class="flex justify-between text-sm font-medium"
      >
        <div
          class="flex min-w-0 pr-2 text-theme-secondary-900 dark:text-theme-dark-50"
        >
          <span
            class="mr-1 truncate"
          >
            123
          </span>
          <span
            class="dark:text-theme-dark-300"
          >
            TEST
          </span>
        </div>
        <span
          class="text-theme-secondary-700 dark:text-theme-dark-200"
        >
          50%
        </span>
      </div>
      <div
        class="h-2 bg-theme-primary-200 dark:bg-theme-dark-600"
      >
        <div
          class="h-2 bg-theme-primary-600"
          style="width: 50%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TokenBalance > should render in xl screen 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col space-y-4 px-6 py-6 sm:px-8"
  >
    <div
      class="flex flex-col items-center space-y-4 sm:flex-row sm:items-end sm:space-y-0"
    >
      <div
        class="w-full sm:w-2/3"
      >
        <div
          class="space-y-0.5 text-center font-medium sm:text-left"
        >
          <p
            class="text-sm leading-5.5 text-theme-secondary-500 dark:text-theme-dark-400"
          >
            My Balance
          </p>
          <p
            class="truncate text-2xl text-theme-secondary-900 dark:text-theme-dark-50"
          >
            $100.00
          </p>
        </div>
      </div>
      <div
        class="sm:w-1/3"
      >
        <div
          class="flex flex-shrink-0 items-center justify-end space-x-3"
        >
          <span
            data-testid="TokenActions__send"
            tabindex="-1"
          >
            <button
              aria-label="Send"
              class="group button-icon-primary"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowUp"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.9 19V9.9h3.8c.3 0 .4-.2.2-.5l-5.7-8.3c-.1-.1-.3-.1-.4 0L4.1 9.4c-.2.4-.1.6.2.6h3.8v3.1"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
          <span
            data-testid="TokenActions__receive"
            tabindex="-1"
          >
            <button
              aria-label="Receive"
              class="group button-icon"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowDown"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.104 1v9.1h-3.8c-.3 0-.4.2-.2.5l5.7 8.3c.1.1.3.1.4 0l5.7-8.3c.2-.4.1-.6-.2-.6h-3.8V6.9"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col space-y-3"
    >
      <div
        class="flex justify-between text-sm font-medium"
      >
        <div
          class="flex min-w-0 pr-2 text-theme-secondary-900 dark:text-theme-dark-50"
        >
          <span
            class="mr-1 truncate"
          >
            123
          </span>
          <span
            class="dark:text-theme-dark-300"
          >
            TEST
          </span>
        </div>
        <span
          class="text-theme-secondary-700 dark:text-theme-dark-200"
        >
          50%
        </span>
      </div>
      <div
        class="h-2 bg-theme-primary-200 dark:bg-theme-dark-600"
      >
        <div
          class="h-2 bg-theme-primary-600"
          style="width: 50%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TokenBalance > should render in xs screen 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col space-y-4 px-6 py-6 sm:px-8"
  >
    <div
      class="flex flex-col items-center space-y-4 sm:flex-row sm:items-end sm:space-y-0"
    >
      <div
        class="w-full sm:w-2/3"
      >
        <div
          class="space-y-0.5 text-center font-medium sm:text-left"
        >
          <p
            class="text-sm leading-5.5 text-theme-secondary-500 dark:text-theme-dark-400"
          >
            My Balance
          </p>
          <p
            class="truncate text-2xl text-theme-secondary-900 dark:text-theme-dark-50"
          >
            $100.00
          </p>
        </div>
      </div>
      <div
        class="sm:w-1/3"
      >
        <div
          class="flex flex-shrink-0 items-center justify-end space-x-3"
        >
          <span
            data-testid="TokenActions__send"
            tabindex="-1"
          >
            <button
              aria-label="Send"
              class="group button-icon-primary"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowUp"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.9 19V9.9h3.8c.3 0 .4-.2.2-.5l-5.7-8.3c-.1-.1-.3-.1-.4 0L4.1 9.4c-.2.4-.1.6.2.6h3.8v3.1"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
          <span
            data-testid="TokenActions__receive"
            tabindex="-1"
          >
            <button
              aria-label="Receive"
              class="group button-icon"
              data-testid="IconButton"
              type="button"
            >
              <svg
                class="h-4 w-4"
                data-testid="icon-FatArrowDown"
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.104 1v9.1h-3.8c-.3 0-.4.2-.2.5l5.7 8.3c.1.1.3.1.4 0l5.7-8.3c.2-.4.1-.6-.2-.6h-3.8V6.9"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col space-y-3"
    >
      <div
        class="flex justify-between text-sm font-medium"
      >
        <div
          class="flex min-w-0 pr-2 text-theme-secondary-900 dark:text-theme-dark-50"
        >
          <span
            class="mr-1 truncate"
          >
            123
          </span>
          <span
            class="dark:text-theme-dark-300"
          >
            TEST
          </span>
        </div>
        <span
          class="text-theme-secondary-700 dark:text-theme-dark-200"
        >
          50%
        </span>
      </div>
      <div
        class="h-2 bg-theme-primary-200 dark:bg-theme-dark-600"
      >
        <div
          class="h-2 bg-theme-primary-600"
          style="width: 50%;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
